src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3739,6 +3739,15 @@ const char* InstanceKlass::internal_name() const {
 void InstanceKlass::print_class_load_logging(ClassLoaderData* loader_data,
                                              const ModuleEntry* module_entry,
                                              const ClassFileStream* cfs) const {
+  if (TraceClassLoadingCause != nullptr &&
+      (strcmp(TraceClassLoadingCause, "*") == 0 || strstr(external_name(), TraceClassLoadingCause) != nullptr))
+  {
+    stringStream st;
```

Comment:
Please use `LogMessage` and `NonInterleavingLogStream` when this is replaced with UL.  See `test_logstream.cpp:117` for usage example.